foreshadow/tests/test_transformers/test_smart/test_smart.py

```diff
@@ -3,6 +3,86 @@
 from foreshadow.utils.testing import get_file_path
 
 
+@pytest.fixture()
+def smart_child():
+    """Get a defined SmartTransformer subclass, TestSmartTransformer.
+
+    Note:
+        Always returns StandardScaler.
+
+    """
+    from foreshadow.smart import SmartTransformer
+    from foreshadow.concrete import StandardScaler
+
+    class TestSmartTransformer(SmartTransformer):
+        def pick_transformer(self, X, y=None, **fit_params):
+            return StandardScaler()
+
+    yield TestSmartTransformer
+
+
+@pytest.mark.parametrize("deep", [True, False])
+def test_smart_get_params_default(smart_child, deep):
+    """Ensure that default get_params works.
+
+    Args:
+        smart_child: a smart instance
+        deep: deep param to get_params
+
+    """
+    smart = smart_child()
+    params = smart.get_params(deep=deep)
```

Comment:
In this test case, deep = True or False doesn't mean any difference. Could you add a case with a fake/mock column_sharer or transformer that can trigger the recursive call? If this is already handled somewhere else, please ignore this comment.